### ISSUE TITLE
Don't clean-up failed jail builds

### ISSUE THREAD

**Issue by igalic** [freebsd/poudriere#996]
## Prerequisites
- [x] Have you checked for an existing issue describing your idea?

## What is your proposal?

Add a flag or config option to stop poudriere from running the cleanup procedures on failure to build a jail

## What is the existing behavior, if any?

When building a poudriere jail from scratch with -B, and something fails, poudriere throws away hours and hours of work, even tho i could easily fix it, and restart it.

## What is the motivation / use case for the change?

I don't wanna waste CPU time, and my own time, rebuilding the same jail because of a transient failure

## Did you consider any alternatives?

The alternative would be to not use poudriere for building PkgBase, but that would take away a lot of flexibility.

## Additional context

Four hours of work down the drain:

```
install: /usr/obj/poudriere/jails/14-current-amd64/usr/src/amd64.amd64/worldstage/usr/share/doc/ntp/ntp.keys.html: chmod: No space left on device
…
[03:54:49] Removing 14-current-amd64 jail... done
[03:55:09] Cleaning 14-current-amd64 data... done
```

this is wasteful, and infuriating.

**Comment by bdrewery** [comment]
Agreed that it is frustrating. Agreed that when building it makes sense to not auto cleanup. However the src build by default does a `make clean` before building. So it too would throw away the progress. `-DNO_CLEAN` is not reliable. `META_MODE` is better and doesn't clean first but isn't default and a bit more complex.

I suppose we need a flag to pass `-DNO_CLEAN` and not cleanup on failure?

**Comment by igalic** [comment]
I do indeed have `NO_CEAN`, `META_MODE`, and long as the obj dir isn't removed, it should rebuild *relatively* quickly
